src/Groups/matrices/linear_centralizer.jl

```diff
@@ -0,0 +1,385 @@
+
+########################################################################
+#
+# Orders
+#
+########################################################################
+
+function _GL_order(n::Int, q::fmpz)
+   res = q^div(n*(n-1),2)
+   for i in 1:n res *= (q^i-1) end
+   return res
+end
+
+_GL_order(n::Int, F::Ring) = _GL_order(n, order(F))
+
+function _SL_order(n::Int, q::fmpz)
+   res = q^div(n*(n-1),2)
+   for i in 2:n res *= (q^i-1) end
+   return res
+end
+
+_SL_order(n::Int, F::Ring) = _SL_order(n, order(F))
+
+########################################################################
+#
+# Centralizer in GL
+#
+########################################################################
+
+# Compute a generating set for every element in GL(n,q) and SL(n,q)
+# generators for GL(n,q) and SL(n,q) are described in:  Taylor, D. E., Pairs of Generators for Matrix Groups. I, The Cayley Bulletin, 3 (1987)
+
+# returns as matrices
+function _gens_for_GL(n::Int, F::Ring)
```

Comment:
This is only valid for F a finite field, right? So perhaps like this?
```suggestion
function _gens_for_GL(n::Int, F::FinField)
```

Reply:
Yes, better to restrict to finite fields.